vnext/Chakra/ChakraJsiRuntime_core.cpp

```diff
@@ -0,0 +1,318 @@
+  }
+
+  JsValueRef result;
+  checkException(JsRun(sourceRef, 0, sourceURLRef, JsParseScriptAttributes::JsParseScriptAttributeNone, &result), sourceURL.c_str());
+}
+
+bool ChakraJsiRuntime::evaluateSerializedScript(const jsi::Buffer& scriptBuffer, const jsi::Buffer& serializedScriptBuffer, const std::string& sourceURL) {
+  JsValueRef bytecodeArrayBuffer = nullptr;
+  if (JsCreateExternalArrayBuffer(const_cast<uint8_t *>(serializedScriptBuffer.data()), serializedScriptBuffer.size(), nullptr, nullptr, &bytecodeArrayBuffer) == JsNoError) {
+
+    JsValueRef sourceURLRef = nullptr;
+    if (!sourceURL.empty()) {
+      sourceURLRef = CreateJSString(reinterpret_cast<const char*>(sourceURL.c_str()), sourceURL.size());
+    }
+
+    JsValueRef value = nullptr;
+    JsErrorCode result = JsRunSerialized(bytecodeArrayBuffer, [](JsSourceContext sourceContext, JsValueRef *value, JsParseScriptAttributes *parseAttributes)
+    {
+      const jsi::Buffer* scriptSource = reinterpret_cast<const jsi::Buffer*>(sourceContext);
+      if (JsCreateExternalArrayBuffer(const_cast<uint8_t *>(scriptSource->data()), scriptSource->size(), nullptr, nullptr, value) != JsNoError)
+        std::terminate();
+
+      *parseAttributes = JsParseScriptAttributeNone;
+      return true;
+    }, reinterpret_cast<JsSourceContext>(pinnedScripts_.back().get()), sourceURLRef, &value);
+
+    if (result == JsNoError) {
+      return true;
+    }
+    else if (result == JsErrorBadSerializedScript) {
+      return false;
+    }
+    else {
+      checkException(result);
+    }
+  }
+
+  return false;
+}
+
+std::unique_ptr<const jsi::Buffer> ChakraJsiRuntime::generatePreparedScript(const std::string& sourceURL, const jsi::Buffer& sourceBuffer) noexcept {
+  const std::wstring scriptUTF16 = facebook::react::UnicodeConversion::Utf8ToUtf16(reinterpret_cast<const char*>(sourceBuffer.data()), sourceBuffer.size());
+
+  unsigned int bytecodeSize = 0;
+  if (JsSerializeScript(scriptUTF16.c_str(), nullptr, &bytecodeSize) == JsNoError)
+  {
+    std::unique_ptr<ByteArrayBuffer> bytecodeBuffer(std::make_unique<ByteArrayBuffer>(bytecodeSize));
+    if (JsSerializeScript(scriptUTF16.c_str(), bytecodeBuffer->data(), &bytecodeSize) == JsNoError)
+    {
+      return bytecodeBuffer;
+    }
+  }
+
+  return nullptr;
+}
+
+
+JsValueRef ChakraJsiRuntime::CreateJSString(const char*data, size_t length) {
+  JsValueRef value;
+  JsCreateString(reinterpret_cast<const char*>(data), length, &value);
+  return value;
+}
+
+JsValueRef ChakraJsiRuntime::CreateJSPropertyId(const char*data, size_t length) {
+  JsValueRef propIdRef;
+  if (JsNoError != JsCreatePropertyId(data, length, &propIdRef)) std::terminate();
+  return propIdRef;
+}
+
+std::wstring ChakraJsiRuntime::JSStringToSTLWString(JsValueRef str) {
+  std::wstring result;
+  size_t length;
+  JsCopyStringUtf16(str, 0, 256 /*TODO*/, nullptr, &length);
+  result.resize(length);

```

Comment:
In the original location we were calling JsGetStringLength, does that not work?  256 seems too small of a max str length here

Reply:
or use JsStringToPointer

---
In reply to: [280988285](https://github.com/Microsoft/react-native-windows/pull/2413#discussion_r280988285) [](ancestors = 280988285)

Reply:
Thanks! Using JsStringToPointer in non-edge mode too.. unifying the implementation.